src/ispd/gui/UserConfigurationDialog.java

```diff
@@ -141,22 +141,21 @@ private String translate(final String word) {
     }
 
     private void onAddClick(final ActionEvent evt) {
-         String add="";
-         add = JOptionPane.showInputDialog(
+
+        final String add = JOptionPane.showInputDialog(
                 this,
                 this.translate("Enter the name"),
                 this.translate("Add"),
                 JOptionPane.QUESTION_MESSAGE
         );
 
         // TODO: Parse result and error.
-        String sResult =JOptionPane.showInputDialog(this,
+        final String sResult =JOptionPane.showInputDialog(this,
                         "Enter user power comsumption limit");
 
 
-        if (add != null && sResult != null && !add.isEmpty() && !this.userSet.contains(add)  && !sResult.isEmpty() && !this.userSet.contains(sResult) ){
-            final Double result =
-                    Double.parseDouble(sResult);
+        if (!add.isEmpty() && !this.userSet.contains(add)  && !sResult.isEmpty() && !this.userSet.contains(sResult) ){
```

Comment:
Have you tested what happens when you cancel the input (i.e., close the prompt window)? Make sure it is working as intended.